src/fsharp/vs/service.fs

```diff
@@ -2352,49 +2357,23 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
             foregroundTypeCheckCount <- foregroundTypeCheckCount + 1
             parseCacheLock.AcquireLock (fun ltok -> 
                 parseAndCheckFileInProjectCachePossiblyStale.Set(ltok, (filename,options),(parseResults,typedResults,fileVersion))  
-                parseAndCheckFileInProjectCache.Set(ltok, (filename,source,options),(parseResults,typedResults,fileVersion,priorTimeStamp))
-                parseFileInProjectCache.Set(ltok, (filename,source,options),parseResults))
```

Comment:
It's possible some of the tests are failing because parseFileInProjectCache is not being populated at this point.  Is it possible to populate it?  I guess that would involve calling ``GetParsingOptionsFromProjectOptions`` which is costly, defeating the purpose of the cache

BTW note that ``parseFileInProjectCache`` should now be renamed to ``parseFileCache`` please

Reply:
I think it would be a good idea to store parsing options like it is currently done for project options and (probably?) pass it here as well.